### PR TITLE
feat: add timers util

### PR DESCRIPTION
**Pull request by squattingmonk**
This timers utility has been backported from nwn-core-framework. When merged, the framework will be updated to use this version.

### CODE REVIEW

**Review by tinygiant98 — COMMENTED** [review]
Ran multiple tests and everything seems to be good to go with the various warning and edge cases being caught.  Haven't tested any `sHandler` uses yet, but it appears that it would work fine once integrated into the framework.

File: src/util_i_timers.nss [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+    if (nTimerID > 0)
+        Notice("Created timer " + IntToString(nTimerID) + sDebug);
+
+    return nTimerID;
+}
+
+int GetIsTimerValid(int nTimerID)
+{
+    // Timer IDs less than or equal to 0 are always invalid.
+    if (nTimerID < 0)
```
[the remark this answers is not in the document]
Comment (squattingmonk) [reply src/util_i_timers.nss:380]: Good catch. Thanks!

File: src/util_i_timers.nss [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+            Error(sDebug + " failed: timer id does not exist");
+    }
+}
+
+void StopTimer(int nTimerID)
+{
+    CreateTimersTable();
+    sqlquery q = SqlPrepareQueryModule(
+        "UPDATE timers SET running = 0 " +
+        "WHERE timer_id = @timer_id RETURNING 1;");
```
[the remark this answers is not in the document]
Comment (squattingmonk) [reply src/util_i_timers.nss:418]: SQLite doesn't normally return anything on an insert, update, or delete, so
`SqlStep()` would return `FALSE` regardless of whether the query did anything or
not. This line makes it return 1 if the update actually changed anything, which
will cause `SQLStep()` to return `TRUE`.

File: src/util_i_timers.nss [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+        Notice("ResetTimer(" + IntToString(nTimerID) + ") successful: " +
+                IntToString(SqlGetInt(q, 0)) + " iterations remaining");
+    }
+}
+
+void KillTimer(int nTimerID)
+{
+    CreateTimersTable();
+    sqlquery q = SqlPrepareQueryModule(
+        "DELETE FROM timers WHERE timer_id = @timer_id RETURNING 1;");
```
[the remark this answers is not in the document]
Comment (squattingmonk) [reply src/util_i_timers.nss:442]: Same as above.
